iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergMajorCompaction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive.compaction;
+
+import java.io.IOException;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.DriverUtils;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.hive.ql.txn.compactor.CompactorContext;
+import org.apache.hadoop.hive.ql.txn.compactor.QueryCompactor;
+import org.apache.iceberg.mr.hive.HiveIcebergMetaHook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergMajorCompaction extends QueryCompactor  {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergMajorCompaction.class.getName());
+
+  @Override
+  public boolean run(CompactorContext context) throws IOException, HiveException, InterruptedException {
+
+    String tableName = context.getTable().getTableName();
+    LOG.debug("Executing compaction query for table {}", tableName);
+
+    String insertIntoTempIcebergTable = String.format("insert into %s select * from %s", tableName, tableName);
```

Comment:
Small query, Since compaction may take some time to complete(for large tables), if there is any parallel writes to this table it will create a new snapshot right? How do we determine the correct snapshot to use(Do we lock table to prevent parallel writes)?

Reply:
We don't lock the table for writes, but if there are parallel writes, only one of the writes will succeed to commit.